internal/xds/xdsclient/xdsresource/xdsconfig.go

```diff
@@ -0,0 +1,111 @@
+/*
+ *
+ * Copyright 2025 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package xdsresource
+
+import "google.golang.org/grpc/resolver"
+
+// XDSConfig holds the complete gRPC client-side xDS configuration containing
+// all necessary resources.
+type XDSConfig struct {
+	// Listener holds the listener configuration. It is guaranteed to be
+	// non-nil.
+	Listener *ListenerUpdate
+
+	// RouteConfig holds the route configuration. It will be populated even if
+	// the route configuration was inlined into the Listener resource. It is
+	// guaranteed to be non-nil.
+	RouteConfig *RouteConfigUpdate
+
+	// VirtualHost selected from the route configuration whose domain field
+	// offers the best match against the provided dataplane authority. It is
+	// guaranteed to be non-nil.
+	VirtualHost *VirtualHost
+
+	// Clusters is a map from cluster name to its configuration.
+	Clusters map[string]*ClusterResult
+}
+
+// ClusterResult contains a cluster's configuration when we receive a valid
+// resource from the management server. It contains an error when:
+//   - we receive an invalid resource from the management server and
+//     we did not already have a valid resource or
+//   - the cluster resource does not exist on the management server
+type ClusterResult struct {
+	Config ClusterConfig
+	Err    error
+}
+
+// ClusterConfig contains configuration for a single cluster.
+type ClusterConfig struct {
+	// Cluster configuration for the cluster. This field is always set to a
+	// non-zero value
+	Cluster *ClusterUpdate
+	// Endpoint configuration for cluster. This field is only set if the cluster
+	// is a leaf cluster.
+	EndpointConfig *EndpointConfig
+	// AggregateConfig is the set of leaf clusters for the cluster. This field
+	// is only set if the cluster is of type AGGREGATE.
+	AggregateConfig *AggregateConfig
+}
+
+// AggregateConfig holds the configuration for an aggregate cluster.
+type AggregateConfig struct {
+	// LeafClusters specifies the names of the leaf clusters for the cluster.
```

Comment:
```suggestion
	// LeafClusters contains a prioritized list of underlying cluster names.
```

Here, and above, we *cannot* assume that an aggregate cluster only contains leaf clusters. Aggregate clusters *may* contain other aggregate clusters too.

The list of child clusters in an aggregate is prioritized. This is an important detail.